### DISCUSSION TITLE
Diffence score F1 between PADIM and me

### DISCUSSION THREAD

**Discussion by davy-blavette** [open-edge-platform/anomalib#2304] [Q&A]
I have a difference in the calculation of my F1 between what PADIM returns to me and my calculation. I don't know how it comes up with an F1 score of 0.95?

![image](https://github.com/user-attachments/assets/b6165dcd-01a4-4119-90b1-161e9da6c4c0)

**Comment by davy-blavette** [comment]
its' ok with         threshold = 0.5...

**Reply by blaz-r** [reply]
This is probably the reason then. Precision and recall depend on the threshold, and anomalib finds the optimal threshold that maximizes the F1 score, then normalizes the predictions in a way that the threshold is at 0.5.

**Reply by davy-blavette** [reply]
@blaz-r yes, I have a different threfold which I calculate by excluding the most good and keeping the most bad. This gives me a SEA score. I calculate it myself, but perhaps you have something similar?

**Reply by samet-akcay** [reply]
@davy-blavette, @blaz-r is right. Anomalib returns `F1_Max` score, based on an adaptive threshold that is computed on the validation set.
